### PR TITLE
chore: add configure scripts and update CI

### LINKED ISSUE #431: Rust cache does not seem working on Windows
I was wondering about the long build times on Windows, and it probably has something to do with symlinks not working on Windows.

https://github.com/pola-rs/r-polars/blob/2fc7ff5e5294f9f4fc9d9b8c3ee8a0254476e4cf/src/Makevars.win#L47-L52

### PR DESCRIPTION
**Pull request by eitsupi**
Part of #408, close #431

- Move some logics on Makevars to the configure script
- Rewrite CI workflows
- Drop R 4.1.x support (for not to use old R on Windows)

Port the configure scripts and GitHub Actions workflows from <https://github.com/eitsupi/prqlr>.

### CODE REVIEW

**Comment by eitsupi** [comment]
The release workflow seems to be building twice. It needs to be fixed.

**Comment by eitsupi** [comment]
Merging this PR should reduce CI execution time due to Rust caching.
I would like to do a major rewrite of the CI after this is merged.

File: .github/actions/setup/action.yaml [hunk 1]
```diff
@@ -5,9 +5,8 @@ inputs:
     description: Set up Rust nightly toolchain?
     required: false
   target:
-    description: Rust toolchain to use
-    required: false
-    default: ""
+    description: Rust target triple to use
```
Comment (etiennebacher) [inline .github/actions/setup/action.yaml:8]: is "target triple" a typo?

File: .github/actions/setup/action.yaml [hunk 2]
```diff
@@ -43,8 +42,26 @@ runs:
         run: |
           rustup target add ${{ inputs.target }}
 
+      - name: Set up Rust pfofile
```
Comment (etiennebacher) [inline .github/actions/setup/action.yaml:45]:
```suggestion
      - name: Set up Rust profile
```

**Review by etiennebacher — COMMENTED** [review]
I don't know much in bash so I can't properly review this but reducing CI time would be great. It looks like it still takes about 20min in this PR, is it because it has to rebuild everything before caching is used?

Also, I didn't understand why we bumped the required R version to 4.2. I'd rather keep it 4.1 if there are no reasons other than for CI.

**Comment by eitsupi** [comment]
> It looks like it still takes about 20min in this PR, is it because it has to rebuild everything before caching is used?

I think that's why it hasn't been saved yet because the cache is only saved from the main branch.
Once this is merged and run once, the build time should be significantly faster as the binaries are cached.

> Also, I didn't understand why we bumped the required R version to 4.2. I'd rather keep it 4.1 if there are no reasons other than for CI.

To support R 4.1, the Makevars file for Windows must be split into `Makevars.win` and `Makevars.ucrt`.
Let's stop supporting R 4.1 because the cost of managing two files is high.

Also, the latest version of libR-sys no longer supports R 4.1 (extendr/libR-sys#170), so it will be necessary to end support for R 4.1 in the near future.

File: .github/actions/setup/action.yaml (hunk 1 above)
Comment (eitsupi) [reply .github/actions/setup/action.yaml:8]: I don't understand the intent of your question. I think this is an official term.
<https://doc.rust-lang.org/beta/rustc/platform-support.html#platform-support>

Comment (etiennebacher) [reply .github/actions/setup/action.yaml:8]: My bad, I just didn't know this expression

**Review by etiennebacher — APPROVED** [review]

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -78,25 +77,32 @@ jobs:
...
-          echo "RPOLARS_RUST_SOURCE=${PWD}/src/rust" >> $GITHUB_ENV
+          if [ "${{ runner.os }}" == "Windows" ]; then
+            TARGET=x86_64-pc-windows-gnu
+          else
+            TARGET="$(rustc -vV | grep host | cut -d' ' -f2)"
+          fi
+          Rscript -e 'pkgbuild::compile_dll()'
+          mkdir -p tools
+          mv "$(pwd)/src/rust/target/${TARGET}/release/libr_polars.a" tools
+          echo "LIBR_POLARS_PATH=$(pwd)tools/libr_polars.a" >>$GITHUB_ENV
```
Comment (eitsupi) [inline .github/workflows/release.yaml:100]:
```suggestion
          echo "LIBR_POLARS_PATH=$(pwd)/tools/libr_polars.a" >>$GITHUB_ENV
```

**Pull request merged**